botocore/auth.py

```diff
@@ -416,7 +416,7 @@ def signature(self, string_to_sign, request):
     def add_auth(self, request):
         if self.credentials is None:
             raise NoCredentialsError()
-        datetime_now = datetime.datetime.utcnow()
+        datetime_now = datetime.datetime.now(datetime.timezone.utc)
```

Comment:
The minimal change here is:

```suggestion
        datetime_now = datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None)
```

The code as it currently stands makes `datetime_now` timezone-aware. The change I'm suggesting above resolves the `DeprecationWarning` without fundamentally altering `datetime_now` in the process.

You can see how `datetime_now` was affected by looking in the test suite, where all of the existing tests had to be modified to suddenly be timezone-aware.